### ISSUE TITLE
User Login fails If sqlite database is locked for writing

### ISSUE THREAD

**Issue by jmbrunskill** [msupply-foundation/open-msupply#3254]
## What went wrong? 😲
User logins fail with message `SOMETHING I CAN"T REMEMBER`

I think this is because SQLite can return a `SQLITE_BUSY` error if we try to write to the db when a write is in progress (aka a transaction)

Our transactions are started with `BEGIN IMMEDIATE` to avoid more than one transaction starting and then getting a deadlock.
```
IMMEDIATE causes the database connection to start a new write immediately, without waiting for a write statement. The BEGIN IMMEDIATE might fail with [SQLITE_BUSY](https://www.sqlite.org/rescode.html#busy) if another write transaction is already active on another database connection.
```

Maybe we can switch back to `DEFERRED` mode? or handle the login errors differently to other error, for example if we can't update the user, should we actually return an error to login? Or just log the error in the logs?
https://www.sqlite.org/lang_transaction.html

## Expected behaviour 🤔

Logins should succeed, or give a useful error messages

## How to Reproduce 🔨

Steps to reproduce the behaviour:

1. Sync a lot of records (e.g. add a new store to you existing site)
2. While sync is running (integrating) try to login

## Your environment 🌱
- Platform: All?
- Database type: SQLite

**Comment by Chris-Petty** [comment]
I think can be high and looked at for 1.7-RC - the risk is a bit unknown, with a site consistently receiving lots of records via sync it could potentially lock regularly. Perhaps it also locks if 3 OMS clients are spamming the API? Perhaps sync buffer integration also affects a user just trying to edit an invoice line? If you have a stocktake with 5000 lines that you're finalising, does it lock for everyone till complete?  

Could potentially use some integration test to validate the bug in one of the above scenarios and then fix.

**Comment by clemens-msupply** [comment]
Problem is that sync needs a long write transaction to integrate all transaction in one go, and you can't read the DB during this time... Switching to DEFERRED mode would IMHO not help here, at least not when sync is going on. So sync integration blocks everything else :(

I guess a message like "Busy, please try later" should definetly help.

We could also further increase the sqlite wait timeout.

What we also can do is make the integration faster: https://github.com/msupply-foundation/open-msupply/issues/2790
This is probably the closed we can get to a fix, i.e. try to integrate all records as quickly as possible to release the write transaction.

**Comment by roxy-dao** [comment]
Closed in #3258
